File: python/composicion.py
import numpy as np

def generar_variables_aleatorias(f, a, b, n):
    # Generar variables aleatorias uniformes entre a y b
    x = np.random.uniform(a, b, n)
    # Aplicar la función f a cada variable aleatoria uniforme
    y = f(x)
    return y

# Ejemplo de uso de la función
f = lambda x: x ** 2 # Función a aplicar
a = 0 # Límite inferior del intervalo
b = 1 # Límite superior del intervalo
n = 10 # Cantidad de números aleatorios a generar

x_aleatorios = generar_variables_aleatorias(f, a, b, n)

print(x_aleatorios)

File: matlab/composicion.m
% Define el número de variables aleatorias a generar
n = 10000;

% Genera dos variables aleatorias uniformes
u1 = rand(n,1);
u2 = rand(n,1);

% Utiliza la transformación inversa para obtener una variable aleatoria normal
x = sqrt(-2*log(u1)).*cos(2*pi*u2);

% Verifica la distribución de la variable aleatoria generada
hist(x,50);

% En este ejemplo, se generan dos variables aleatorias uniformes u1 y u2 utilizando la función rand de Matlab.
%  Luego, se utiliza la transformación inversa de la función de densidad de probabilidad de la distribución normal
%   para obtener la variable aleatoria x con distribución normal. 
%   Finalmente, se utiliza la función hist para verificar que la distribución de la variable aleatoria generada sea similar a 
%   una distribución normal.


